src/url-loader/url-resolver.ts

```diff
@@ -26,6 +26,12 @@ import {FileRelativeUrl, ResolvedUrl} from '../model/url';
  * to '/bower_components/polymer/polymer.html'.
  */
 export abstract class UrlResolver {
+  static async createForDirectory(_dirname: string): Promise<UrlResolver> {
+    // Break the import loop by doing a dynamic import.
+    const {PackageUrlResolver} = await import('./package-url-resolver');
+    return new PackageUrlResolver(/* reference dirname here soon */);
```

Comment:
I feel a little weird that we're adding new API that defaults to using `bower_components` here. I'd rather either:

 1. Default to _no_ component directory, ie, not package-semantics, if the intention here is to be generic.
 2. Require a component directory so we have fewer places to dig up when we want to default to npm settings.

Also, if the only thing this does is delegate to the `PackageUrlResolver`, and that forces it to be async I question the value of this method. Is it that more ergonomic than `new PackageUrlResolver()`?